changelog/2024-10-01T14_41_36+02_00_fix_index_literal_rendering

```diff
@@ -0,0 +1,2 @@
+FIXED: Clash hanging when rendering `Index n` literals, for large values of `n` [#2813](https://github.com/clash-lang/clash-compiler/issues/2813)
+FIXED: Render overflowed Index literals as errors
```

Comment:
```suggestion
FIXED: Render overflowed Index literals as don't-cares in HDL
```
?

Maybe that's not good either, but I worry people might not understand what "render as an error" means. I'd also like to emphasise that we're talking about HDL here for clarity.